#!/usr/bin/env python

PACKAGE = "quadrotor_input"
import roslib;roslib.load_manifest(PACKAGE)
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("z_p", double_t, 0, "Z Pos Proportional Gain", 0, 0.0, 1000.0)
gen.add("z_i", double_t, 0, "Z Pos Integral Gain", 0.0, 0.0, 100.0)
gen.add("z_d", double_t, 0, "Z Pos Derivative Gain", 0.0, 0.0, 100.0)
gen.add("z_i_min", double_t, 0, "Z Pos Integral Min Limit", 0.0, -1000.0, 0.0)
gen.add("z_i_max", double_t, 0, "Z Pos Integral Max Limit", 0.0, 0.0, 1000.0)
gen.add("z_des", double_t, 0, "Z position", 0.2, -2.0, 10.0)
gen.add("base_throttle_start", double_t, 0, "Base Throttle Start", 0, 0, 1.0)
gen.add("voltage_start", double_t, 0, "Voltage Start", 0, 10.0, 20.0)
gen.add("base_throttle_end", double_t, 0, "Base Throttle End", 0, 0, 1.0)
gen.add("voltage_end", double_t, 0, "Voltage End", 0, 10.0, 20.0)
gen.add("mass", double_t, 0, "Aircraft Mass", 2.0, 0.1, 10.0)
gen.add("pid_limit", double_t, 0, "Limit of PID Command (abs)", 0, 0, 1.0)
gen.add("cmd_max", double_t, 0, "Maximum overall throttle", 0, 0, 1.0)
gen.add("cmd_min", double_t, 0, "Minimum overall throttle" , 0, 0, 1.0)

exit(gen.generate(PACKAGE, "quadrotor_input", "AltitudeController"))
